File: python/reverse-string.py
from typing import List


class Solution:
    # Time complexity: O(n) where n is the length of the string
    # Space complexity: O(1)
    def reverseString(self, s: List[str]) -> None:
        """
        Do not return anything, modify s in-place instead.
        """
        for i in range(len(s)//2):
            s[i], s[~i] = s[~i], s[i]

File: ts/reverse-string.ts
/**
 Do not return anything, modify s in-place instead.
 */
// Time complexity: O(n)
// Space complexity: O(1)
const reverseString = function (s: string[]): void {
    for (let i = 0; i < Math.floor(s.length / 2); i++) {
        [s[i], s[s.length - i - 1]] = [s[s.length - i - 1], s[i]]
    }
};